pkg/graveler/graveler.go

```diff
@@ -1745,6 +1778,24 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 	return newCommitID, nil
 }
 
+func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repositoryID RepositoryID, branchID BranchID, f BranchUpdateFunc) error {
+	const setTries = 3
+
+	var try int
+	for try = 0; try < setTries; try++ {
+		// TODO - if the branch commit id hasn't changed, update the fields instead of fail
+		err := g.RefManager.BranchUpdate(ctx, repositoryID, branchID, f)
+		if errors.Is(err, kv.ErrPredicateFailed) {
```

Comment:
I think we should add a log when we retry

Reply:
done